docs/IE_PLUGIN_DG/NewTransformation.md

```diff
@@ -1,15 +1,390 @@
+> **Note**: this operation are prohibited to use inside new plugins until they are not moved to separate directory with allowed operations.
+2. transformations - includes all transformations, utils, runtime info attributes and pass managers.
+> **Note**: do not use transformation that belongs to `ngraph::pass::ConvertOpSet1ToLegacy` transformations until they are not moved to separate directory with allowed transformations.
 
-bla-bla
+Transformation flow in transformation library has several layers:
+1. Pass managers - executes list of transformations using `*_tbl.hpp` file. For example conversion form OpSetX to OpSetY.
+2. Transformations - performs particular transformation algorithm on `ngraph::Function`. Find more about transformations in [Transformations types](#transformations_types).
+3. Low level functions that takes set of nodes and performs some transformation action. They are not mandatory and all transformation code can be located inside transformation. But if some transformation parts can potentially be reused in other transformations we suggest to keep them as a separate functions.
 
-// ! [new_transformation:part1]
+To decide where to store your transformation code please follow these rules:
+1. If it's plugin specific transformation and can't be reused by other plugins keep source code inside plugin.
+2. If this transformation relates to OpSetXToOpSetY conversion or it's common optimization then keep sources inside transformation library.
 
-And this file must refer to that code via
+After you decided where to store your transformation code you can start develop your own nGraph transformation.
 
-@snippet src/template_transformation.cpp new_transformation:part1
+## Table of Contents:
+
+1. [`ngraph::Function` and graph representation](#ngraph_function) 
+2. [Transformations types](#transformations_types)
+3. [Pattern matching](#pattern_matching)
+4. [Working with ngraph::Function](#working_with_ngraph_function)
+5. [Transformation writing essentials](#transformation_writing_essentials)
+6. [Common mistakes in transformations](#common_mistakes)
+7. [Using pass manager](#using_pass_manager)
+8. [How to debug transformations](#how_to_debug_transformations)
+9. [Disabling/Enabling specific transformations for plugin X](#disabling_transformation)
+10. [Transformations testing](#transformations_testing)
+
+## ngraph::Function and graph representation <a name="ngraph_function"></a>
+
+nGraph function is a very simple thing: it stores shared pointers to `ngraph::op::Result` and `ngraph::op::Parameter` operations that are inputs and outputs of the graph. 
+All other operations hold each other via shared pointers: child operation holds its parent (hard link). If operation has no consumers and it's not Result operation
+(shared pointer counter is zero) then it will be destructed and won't be accessible anymore. Each operation in `ngraph::Function` is a shared_ptr and has `ngraph::Node` as a base class.
```

Comment:
maybe just:

Each operation in `ngraph::Function` is a `std::shared_ptr<ngraph::Node>`

?

Reply:
done